src/test/java/seedu/application/model/application/UpcomingInterviewPredicateTest.java

```diff
@@ -41,14 +41,36 @@ public void test_isUpcomingInterview_returnsTrue() {
         // they are considered to have passed
 
         // EP: interview is a minute later
-        assertTrue(predicate.test(new ApplicationBuilder().withInterview(
-                new InterviewBuilder().withInterviewDate(LocalDate.now())
-                        .withInterviewTime(LocalTime.now().plusMinutes(1)).build()).build()));
+        Application aMinuteLaterApplication;
+        //Take account of time is 2359
+        if (LocalTime.now().plusMinutes(1).equals(LocalTime.of(0, 0))) {
+            aMinuteLaterApplication = new ApplicationBuilder().withInterview(
+                    new InterviewBuilder().withInterviewDate(LocalDate.now().plusDays(1))
+                            .withInterviewTime(LocalTime.of(0, 0)).build()).build();
+        } else {
+            aMinuteLaterApplication = new ApplicationBuilder().withInterview(
+                    new InterviewBuilder().withInterviewDate(LocalDate.now())
+                            .withInterviewTime(LocalTime.now().plusMinutes(1)).build()).build();
+        }
```

Comment:
I think this might still go wrong because `LocalTime` also stores seconds, so the 2359 plus one minute may not `equal` midnight. How about using `LocalDateTime` instead?
```suggestion
        LocalDateTime aMinuteLater = LocalDateTime.now().plusMinutes(1);
        Application aMinuteLaterApplication = new ApplicationBuilder().withInterview(
                new InterviewBuilder().withInterviewDate(aMinuteLater.toLocalDate()),
                        .withInterviewTime(aMinuteLater.toLocalTime()).build()).build();
```

Reply:
Thanks for the suggestion! didn't know LocalDateTime can do this 😅